examples/native-glutin/src/main.rs

```diff
@@ -49,30 +57,15 @@ fn main() {
 
     gl.viewport(0, 0, 300, 200);
 
-    let mut running = true;
-    while running {
-        events_loop.poll_events(|event| {
-            // println!("{:?}", event);
-            #[allow(clippy::single_match)]
-            match event {
-                glutin::Event::WindowEvent { event, .. } => match event {
-                    glutin::WindowEvent::CloseRequested => running = false,
-                    glutin::WindowEvent::Resized(logical_size) => {
-                        let dpi_factor = gl_window.get_hidpi_factor();
-                        gl_window.resize(logical_size.to_physical(dpi_factor));
-                    },
-                    _ => (),
-                },
-                _ => ()
-            }
-        });
+    while app.is_running() {
+        app.poll_events();
 
         gl.clear(glenum::BufferBit::Color);
         gl.draw_arrays(gl::TRIANGLES, 0, 3);
 
         // check_error!();
 
-        let _ = gl_window.swap_buffers();
+        app.window(window_id).unwrap().swap_buffers();
```

Comment:
I'm not quite fond of doing a HashMap-lookup every frame when there's usually just a single window.
Possibility: store as `Vec<(WindowId, Window)>`. Index-lookup would be faster, and I'm not sure how often a lookup by Id is required - might be fine to do a `.iter().find(..)`.

Reply:
The main loop itself needs a lookup by id, which happens not only per frame but per event, so possible hundreds of times. I see no way to change this, as the event only gives us the window id.

Reply:
I don't see how to get rid of the hash map at the moment. But I added an optimization so that in the case of only one window, that window is always returned without a hash map uplook. I usually don't like these kinds of optimizations, but in this case I think it is OK, as this will really be the default case